src/modules/core/components/avatar/avatar.tsx

```diff
@@ -0,0 +1,82 @@
+import { type FC, useMemo } from 'react';
+import clsx from 'clsx';
+import Image from 'next/image';
+
+import type { AvatarProps } from './avatar.interface';
+import styles from './avatar.module.scss';
+
+export const Avatar: FC<AvatarProps> = ({
+  url,
+  size = 'small',
+  fallback = <UserIcon />,
+  shadow = false,
+  shape = 'circle',
+}) => {
+  const avatarSize = useMemo(() => {
+    const SIZES = {
+      medium: 56,
+      large: 156,
+      small: 40,
+    };
+
+    if (size === 'medium') {
+      return SIZES.medium;
+    } else if (size === 'large') {
+      return SIZES.large;
+    }
+
+    return SIZES.small;
+  }, [size]);
+
+  return (
+    <div
+      className={clsx(styles.avatarContainer, styles[shape], {
+        [styles.shadow]: shadow,
+      })}
+      style={{
+        width: `${avatarSize}px`,
+        height: `${avatarSize}px`,
+      }}
+    >
+      {url ? (
+        <Image
+          src={url}
+          alt='Avatar'
+          width={avatarSize}
+          height={avatarSize}
+          style={{
+            objectFit: 'cover',
+          }}
+        />
+      ) : (
+        <div className={styles.fallbackContainer}>{fallback}</div>
+      )}
+    </div>
+  );
+};
+
+const UserIcon: FC = () => (
```

Comment:
This is a simple svg, it should be in the `Icon` component, and used via `Icon`, like `<Icon name='profile' />` or something similiar